centreon/src/Core/HostGroup/Infrastructure/Repository/DbReadHostGroupRepository.php

```diff
@@ -106,31 +107,190 @@ public function findNames(array $hostGroupIds): HostGroupNamesById
     {
-        if ([] === $accessGroups) {
+        if ([] === $accessGroupIds) {
             return new \ArrayIterator([]);
         }
 
-        $accessGroupIds = $this->accessGroupsToIds($accessGroups);
+        [$bindValues, $bindQuery] = $this->createMultipleBindQuery($accessGroupIds, ':access_group_id_');
 
-        if ($this->hasAccessToAllHostGroups($accessGroupIds)) {
+        $request = <<<'SQL'
+            SELECT DISTINCT
+                hg.hg_id,
+                hg.hg_name,
+                hg.hg_alias,
+                hg.hg_notes,
+                hg.hg_notes_url,
+                hg.hg_action_url,
+                hg.hg_icon_image,
+                hg.hg_map_icon_image,
+                hg.hg_rrd_retention,
+                hg.geo_coords,
+                hg.hg_comment,
+                hg.hg_activate
+            FROM `:db`.`hostgroup` hg
+            SQL;
+
+        $sqlTranslator = $requestParameters ? new SqlRequestParametersTranslator($requestParameters) : null;
+        $sqlTranslator?->setConcordanceArray([
+            'id' => 'hg.hg_id',
+            'alias' => 'hg.hg_alias',
+            'name' => 'hg.hg_name',
+            'is_activated' => 'hg.hg_activate',
+            'hostcategory.id' => 'hc.hc_id',
+            'hostcategory.name' => 'hc.hc_name',
+        ]);
+        $sqlTranslator?->addNormalizer('is_activated', new BoolToEnumNormalizer());
 
-            return $this->findAll($requestParameters);
+        // Update the SQL string builder with the RequestParameters through SqlRequestParametersTranslator
+        $searchRequest = $sqlTranslator?->translateSearchParameterToSql();
+
+        // Only JOIN if search request has been provided...
+        if ($searchRequest !== null) {
+            $hostCategoryAcls = $this->generateHostCategoryAclSubRequest($accessGroupIds);
+            $request .= <<<SQL
+
+                LEFT JOIN `:db`.hostgroup_relation hgr
+                    ON hgr.hostgroup_hg_id = hg.hg_id
+                LEFT JOIN `:db`.hostcategories_relation hcr
+                    ON hcr.host_host_id = hgr.host_host_id
+                    AND hcr.hostcategories_hc_id IN ({$hostCategoryAcls})
+                LEFT JOIN `:db`.hostcategories hc
+                    ON hc.hc_id = hcr.hostcategories_hc_id
+                    AND hc.level IS NULL
+                SQL;
         }
 
-        $concatenator = $this->getFindHostGroupConcatenator($requestParameters, $accessGroupIds);
+        $request .= <<<'SQL'
+
+            INNER JOIN `:db`.acl_resources_hg_relations arhr
+                ON hg.hg_id = arhr.hg_hg_id
+            INNER JOIN `:db`.acl_resources res
+                ON arhr.acl_res_id = res.acl_res_id
+            INNER JOIN `:db`.acl_res_group_relations argr
+                ON res.acl_res_id = argr.acl_res_id
+            INNER JOIN `:db`.acl_groups ag
+                ON argr.acl_group_id = ag.acl_group_id
+            SQL;
```

Comment:
Why didn't you integrate these joins directly into the query?

Reply:
yep you are right